vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs

```diff
@@ -726,4 +728,216 @@ type internal FSharpNavigation
+        // "M:N.X.gg(System.Int16[],System.Int32[0:,0:])" - method with two parameters, 1d and 2d array
+        // "M:N.X.op_Addition(N.X,N.X)" - operator
+        // "M:N.X.op_Explicit(N.X)~System.Int32" - operator with return type
+        // "M:N.GenericMethod.WithNestedType``1(N.GenericType{``0}.NestedType)" - generic type with one parameter
+        // "M:N.GenericMethod.WithIntOfNestedType``1(N.GenericType{System.Int32}.NestedType)" - generic type with one parameter
+        // "M:N.X.N#IX{N#KVP{System#String,System#Int32}}#IXA(N.KVP{System.String,System.Int32})" - explicit interface implementation
+        //
+        // "E:" prefix for events
+        //
+        // "E:N.X.d".
+        // 
+        // "F:" prefix for fields
+        // "F:N.X.q" - field
+        //
+        // "P:" prefix for properties
+        // "P:N.X.prop" - property with getter and setter
+
+        let m = docCommentIdRx.Match(docId)
+        let t = m.Groups[1].Value
+        match m.Success, t with
+        | true, ("M" | "P" | "E") ->
+            // TODO: Probably, there's less janky way of dealing with those.
+            let parts = m.Groups[2].Value.Split('.')
+            let entityPath = parts[..(parts.Length - 2)] |> List.ofArray
+            let memberOrVal = parts[parts.Length - 1]
+
+            // Try and parse generic params count from the name (e.g. NameOfTheFunction``1, where ``1 is amount of type parameters) 
+            let genericM = fnGenericArgsRx.Match(memberOrVal)
+            let (memberOrVal, genericParametersCount) =
+                if genericM.Success then
+                    (genericM.Groups[1].Value, int genericM.Groups[2].Value)
+                else
+                    memberOrVal, 0
+
+            // A hack/fixup for the constructor name (#ctor in doccommentid and ``.ctor`` in F#)
+            if memberOrVal = "#ctor" then
+                DocCommentId.Member ({ EntityPath = entityPath; MemberOrValName = "``.ctor``"; GenericParameters = 0 }, (SymbolMemberType.FromString "CTOR"))
+            else
+                DocCommentId.Member ({ EntityPath = entityPath; MemberOrValName = memberOrVal; GenericParameters = genericParametersCount }, (SymbolMemberType.FromString t))
+        | true, "T" ->
+            let entityPath = m.Groups[2].Value.Split('.') |> List.ofArray
+            DocCommentId.Type entityPath
+        | true, "F" ->
+            let parts = m.Groups[2].Value.Split('.')
+            let entityPath = parts[..(parts.Length - 2)] |> List.ofArray
+            let memberOrVal = parts[parts.Length - 1]
+            DocCommentId.Field { EntityPath = entityPath; MemberOrValName = memberOrVal; GenericParameters = 0 }
+        | _ -> DocCommentId.None
+    
+    let tryFindFieldByName (name: string) (e: FSharpEntity) =
+        e.FSharpFields
+        |> Seq.filter (
+            fun x ->
+                x.DisplayName = name
+                && not x.IsCompilerGenerated)
```

Comment:
If someone wants to navigate from C# to a compiler generated member (e.g. Is* for DU's), I think we should still enable such jump, e.g. to the apparent enclosing type. Or?

Reply:
For compiler generated it will navigate to decompiled source, it's a fallback option when we don't return anything from here (i.e. return null to Roslyn).

If you want to navigate to DU, just have to do it on type.

Reply:
Once we have an ability to show multiple results - we can give a choice

Reply:
But isn't the natural consumption of DU's from C# via compiler generated code mainly?
I would see it definitely more natural to jump to the DU definition , rather then showing decompiled source in the case of F# project being in the same solution.

Reply:
After all, the context are mixed solutions. Therefore audience are people who at least know F# exists and who do have an F# project in the solution.

I can't imagine a decompiled source being better then jumping to at least an approximate location. (e.g. type in case of compiler generated code)

Reply:
We can do it, yeah.